plugins/BEdita/API/src/Controller/AppController.php

```diff
@@ -177,34 +177,21 @@ protected function html()
 
             $this->components()->unload('JsonApi');
             unset($this->JsonApi);
-            $viewBuilder->template('Common/html');
+            $viewBuilder->setTemplate('Common/html');
         } catch (\Exception $exception) {
             $renderer = new ExceptionRenderer($exception);
             $response = $renderer->render();
-            $responseBody = $response->body();
-            $this->response->statusCode($response->statusCode());
-            $viewBuilder->template('Error/error');
+            $responseBody = (string)$response->getBody();
+            $this->response = $this->response->withStatus($response->getStatusCode());
+            $viewBuilder->setTemplate('Error/error');
         }
 
         $this->set(compact('method', 'responseBody', 'url'));
 
         // render HTML
-        $viewBuilder->className('View');
+        $viewBuilder->setClassName('View');
         $this->response->type('html');
 
         return $this->render();
     }
-
-    /**
-     * Send a 204 No Content response with empty body.
-     *
-     * @return void
-     */
-    protected function noContentResponse()
-    {
-        $this->autoRender = false;
-        $this->response->header('Content-Type: ' . $this->request->contentType());
-        $this->response->statusCode(204);
-        $this->response->send();
```

Comment:
`Response::send()` method has been deprecated. Therefore I moved all this logic to the invoking points, and a Response object is returned.